com.unity.render-pipelines.core/ShaderLibrary/DataExtraction.hlsl

```diff
@@ -0,0 +1,100 @@
+
+int UNITY_DataExtraction_Mode;
+int UNITY_DataExtraction_Space;
+
+#define RENDER_OBJECT_ID               1
+#define RENDER_DEPTH                   2
+#define RENDER_WORLD_NORMALS_FACE_RGB  3
+#define RENDER_WORLD_POSITION_RGB      4
+#define RENDER_ENTITY_ID               5
+#define RENDER_BASE_COLOR_RGBA         6
+#define RENDER_SPECULAR_RGB            7
+#define RENDER_METALLIC_R              8
+#define RENDER_EMISSION_RGB            9
+#define RENDER_WORLD_NORMALS_PIXEL_RGB 10
+#define RENDER_SMOOTHNESS_R            11
+#define RENDER_OCCLUSION_R             12
+#define RENDER_DIFFUSE_COLOR_RGBA      13
+
+
+void ConvertSpecularToMetallic(float3 diffuseColor, float3 specularColor, out float3 baseColor, out float metallic)
+{
```

Comment:
This file should be in universal. HDRP have already its own API (AOV API and Debug API) to extract information and will not rely on this shader.

Side question: Why do we have all the Specular color and Mettalic for SRP Picking. Guess there is a mix here of tooling helper for picking (object id, depth, normal, entity) and function to extract information for LOD. So we mix two different concept. Are we really sure we want to go through the same API?

Reply:
The main thing we need to solve is have a common API across render pipelines. An API that is specific to HDRP is not very useful when writing high level tools. Like Recorder / Imposter / LOD / Picking.

On the backend for HDRP it is perfectly fine if the API is just a wrapper around the AOV / Debug API.

Reply:
Move to CommonMaterial.hlsl